tripal_layout/src/Controller/TripalEntityUILayoutController.php

```diff
@@ -409,7 +418,130 @@ protected function addDetailsFieldGroup($name, EntityDisplayBase $display, $sett
           "description" => $this->getSetting('description', '', $settings),
         ],
       ];
-      $display->setThirdPartySetting('field_group', $name, $details_group);
+      $display->setThirdPartySetting('field_group', $name, $group);
+    }
+  }
+
+  /**
+   * Adds a 'details_sidebar' field group.
+   *
+   * @param string $name
+   *   The name of the field group component.
+   * @param EntityDisplayBase $display
+   *   The display configuration to add the field group component to.
+   * @param array $settings
+   *   An array of settings to apply to the field group.
+   */
+  protected function addDetailsSidebarFieldGroup($name, EntityDisplayBase $display, $settings = []) {
+    $field_groups = $display->getThirdPartySettings('field_group');
+
+    // If the field group doesn't exist then add it.
+    if (!array_key_exists($name, array_keys($field_groups))) {
+
+      $classes = $this->getSetting('classes', '', $settings);
+      $classes .= 'tripal-layout-details-sidebar';
+
+      $parent_name = $this->getSetting('parent_name', '', $settings);
+      $group = [
+        "children" => [],
+        "label" => $this->getSetting('label', 'Missing Label', $settings),
+        "parent_name" => $parent_name,
+        "region" => $this->getSetting('region', 'content', $settings),
+        "weight" => $this->getSetting('weight', 0, $settings),
+        "format_type" => "details_sidebar",
+        "format_settings" => [
+          "classes" => $classes,
+          "show_empty_fields" => $this->getSetting('show_empty', FALSE, $settings),
+          "id" => $name,
+          "open" => $this->getSetting('open', FALSE, $settings),
+          "description" => $this->getSetting('description', '', $settings),
+          "label_as_html" => $this->getSetting('label_as_html', FALSE, $settings),
+          "required_fields" => $this->getSetting('required_fields', TRUE, $settings),
+          "weight" => $this->getSetting('weight', 0, $settings),
+        ],
+      ];
+      $display->setThirdPartySetting('field_group', $name, $group);
+    }
+  }
+
+  /**
+   * Adds a 'tabs' field group.
+   *
+   * @param string $name
+   *   The name of the field group component.
+   * @param EntityDisplayBase $display
+   *   The display configuration to add the field group component to.
+   * @param array $settings
+   *   An array of settings to apply to the field group.
+   */
+  protected function addVerticalTabsFieldGroup($name, EntityDisplayBase $display, $settings = []) {
+    $field_groups = $display->getThirdPartySettings('field_group');
+
+    // If the field group doesn't exist then add it.
+    if (!array_key_exists($name, array_keys($field_groups))) {
```

Comment:
I happened to notice that this may not do what you expect, you seem to be changing array keys to array elements before looking for the key, so the `array_keys()` part should be removed.
Also lines 439 and 521 and 566 same thing.
It may not have shown up functionally because the array keys indeed did not exist when I did the PR testing as described, and reapplying tripal layout behaves the same way as the first time.

Reply:
Since I have already tested this change, I just pushed this change to save you time 😁

Reply:
Thanks!